rust/kcl-lib/src/std/planes.rs

```diff
@@ -1,15 +1,96 @@
 
-use super::{args::TyF64, sketch::PlaneData};
+use super::{
+    args::TyF64,
+    sketch::{FaceTag, PlaneData},
+};
 use crate::{
-    errors::KclError,
-    execution::{ExecState, KclValue, ModelingCmdMeta, Plane, PlaneType, types::RuntimeType},
+    UnitLen,
+    errors::{KclError, KclErrorDetails},
+    execution::{ExecState, KclValue, Metadata, ModelingCmdMeta, Plane, PlaneType, types::RuntimeType},
     std::Args,
 };
 
+/// Find the plane of a given face.
+pub async fn plane_of(exec_state: &mut ExecState, args: Args) -> Result<KclValue, KclError> {
+    let solid = args.get_unlabeled_kw_arg("solid", &RuntimeType::solid(), exec_state)?;
+    let face = args.get_kw_arg("face", &RuntimeType::tagged_face(), exec_state)?;
+
+    inner_plane_of(solid, face, exec_state, &args)
+        .await
+        .map(Box::new)
+        .map(|value| KclValue::Plane { value })
+}
+
+async fn inner_plane_of(
+    solid: crate::execution::Solid,
+    face: FaceTag,
+    exec_state: &mut ExecState,
+    args: &Args,
+) -> Result<Plane, KclError> {
+    // Query the engine to learn what plane, if any, this face is on.
+    let face_id = face.get_face_id(&solid, exec_state, args, true).await?;
+    let meta = args.into();
+    let cmd = ModelingCmd::FaceIsPlanar(mcmd::FaceIsPlanar { object_id: face_id });
+    let plane_resp = exec_state.send_modeling_cmd(meta, cmd).await?;
+    let OkWebSocketResponseData::Modeling {
+        modeling_response: OkModelingCmdResponse::FaceIsPlanar(planar),
+    } = plane_resp
+    else {
+        return Err(KclError::new_semantic(KclErrorDetails::new(
+            format!(
+                "Engine returned invalid response, it should have returned FaceIsPlanar but it returned {plane_resp:#?}"
+            ),
+            vec![args.source_range],
+        )));
+    };
+    // Destructure engine's response to check if the face was on a plane.
+    let not_planar: Result<_, KclError> = Err(KclError::new_semantic(KclErrorDetails::new(
+        "The face you provided doesn't lie on any plane. It might be curved.".to_owned(),
+        vec![args.source_range],
+    )));
+    let Some(x_axis) = planar.x_axis else { return not_planar };
+    let Some(y_axis) = planar.y_axis else { return not_planar };
+    let Some(origin) = planar.origin else { return not_planar };
+    // TODO: Check this is accurate. I think the engine always returns
+    // millimeters so it's correct.
+    let units = UnitLen::Mm;
+    let x_axis = crate::execution::Point3d {
+        x: x_axis.x,
+        y: x_axis.y,
+        z: x_axis.z,
+        units,
+    };
+    let y_axis = crate::execution::Point3d {
+        x: y_axis.x,
+        y: y_axis.y,
+        z: y_axis.z,
+        units,
+    };
+    let origin = crate::execution::Point3d {
+        x: origin.x.0,
+        y: origin.y.0,
+        z: origin.z.0,
+        units,
+    };
+
+    // Engine doesn't send back an ID, so let's just make a new plane ID.
```

Comment:
It'd be great if the engine would send back a plane ID in the future, so that we don't need to set Uninit (line 86) and avoid another API call in the future.